master/internal/stream/system.go

```diff
@@ -109,22 +110,38 @@ func (ps PublisherSet) Start(ctx context.Context) {
 	// go publishLoop(ctx, "stream_experiment_chan", ps.Experiments)
 }
 
-func writeAll(socket *websocket.Conn, msgs []*websocket.PreparedMessage) error {
+func writeAll(socketLike api.SocketLike, msgs []interface{}) error {
 	for _, msg := range msgs {
-		err := socket.WritePreparedMessage(msg)
-		if err != nil {
-			return err
+		if socketLike.Target != nil {
+			err := socketLike.Target.Send(msg)
+			if err != nil {
+				return err
+			}
+		} else {
+			pm, ok := msg.(*websocket.PreparedMessage)
+			if !ok {
+				return fmt.Errorf("received message that is not a prepared message")
+			}
+			err := socketLike.WritePreparedMessage(pm)
+			if err != nil {
+				return err
+			}
 		}
 	}
 	return nil
 }
 
+//type FakePreparedMessage struct {
```

Comment:
question: is this meant to be here? Are you looking at replacing the interface{}?